konrad/humidity.py

```diff
@@ -204,21 +208,19 @@ def omega(self):
     def get_relative_humidity_profile(self, p):
         return self.rh_surface * ((p / p[0] - 0.02) / (1 - 0.02))**self.omega
 
-    def get(self, plev, T, z, T_surface=288, **kwargs):
+    def get(self, atmos, **kwargs):
         """Determine the humidity profile based on atmospheric state.
 
         Parameters:
-            plev (ndarray): Pressure levels [Pa].
-            T (ndarray): Temperature [K].
-            z (ndarray): Height [m].
-            T_surface (float): Surface temperature [K].
+            atmos (konrad.atmosphere): Atmosphere object,
+                including temperature, pressure, altitude, surface
 
         Returns:
             ndarray: Water vapor profile [VMR].
         """
-        self.T_surface = T_surface
+        self.T_surface = atmos.surface['temperature'].values[-1]
```

Comment:
In a next step, we could also detach the `surface` form the `atmosphere`.

The core could for example look like:
```python
humidity.get(self.atmosphere, self.surface, heatingrates)
```

But that's up to you 😉

Reply:
To elaborate on this: All code from `Atmosphere.adjust()` could eventually be called in `RCE.run()`